<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShLoop</class>
 <widget class="QDialog" name="ShLoop">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>522</width>
    <height>617</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Loop</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff/>
   </iconset>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Condition</string>
      </attribute>
      <layout class="QGridLayout">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="fContent">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="margin">
           <number>0</number>
          </property>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tStatistics">
      <attribute name="title">
       <string>Statistics</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="margin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QSplitter" name="splitter">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <widget class="QTableView" name="tvStatTable">
          <property name="selectionMode">
           <enum>QAbstractItemView::NoSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="horizontalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="showGrid">
           <bool>false</bool>
          </property>
          <property name="sortingEnabled">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QWidget" name="fCurves" native="true">
          <layout class="QVBoxLayout" name="_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>0</number>
             </property>
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QCheckBox" name="cbActive">
               <property name="text">
                <string>Active</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="cbDone">
               <property name="text">
                <string>Done</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="cbOut">
               <property name="text">
                <string>Out</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="CurveView" name="fCurvesDisplay">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>30</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="margin">
               <number>0</number>
              </property>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="lTotal">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Total:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lTotalCount">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="lActive">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Active:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lActiveCount">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="lDone">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Done:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lDoneCount">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="lOut">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Out:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lOutCount">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="pbBreak">
         <property name="text">
          <string>Break</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbStatistics">
         <property name="text">
          <string>Full statistic</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbJump">
         <property name="text">
          <string>Jump</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbNext">
         <property name="text">
          <string>Next</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QFrame" name="fProgress">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="pbStopProgress">
          <property name="toolTip">
           <string>Stop processing</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../glslDevil.qrc">
            <normaloff>:/icons/icons/process-stop_32.png</normaloff>:/icons/icons/process-stop_32.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="pbProgress">
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="format">
           <string>%v/%m</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="fJump">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="pbStopJump">
          <property name="toolTip">
           <string>Stop processing</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../glslDevil.qrc">
            <normaloff>:/icons/icons/process-stop_32.png</normaloff>:/icons/icons/process-stop_32.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="hsJump">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::NoTicks</enum>
          </property>
          <property name="tickInterval">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lJump">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="spacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>2</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CurveView</class>
   <extends>QFrame</extends>
   <header>dialogs/curveview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../glslDevil.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>hsJump</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lJump</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>344</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>485</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbStatistics</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>calculateStatistics()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbJump</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>doJump()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbNext</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>nextIteration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbBreak</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>doBreak()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbStopProgress</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>stopProgress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>16</x>
     <y>562</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbStopJump</sender>
   <signal>clicked()</signal>
   <receiver>ShLoop</receiver>
   <slot>stopJump()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>16</x>
     <y>596</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>308</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>calculateStatistics()</slot>
  <slot>nextIteration()</slot>
  <slot>doJump()</slot>
  <slot>doBreak()</slot>
  <slot>stopProgress()</slot>
  <slot>stopJump()</slot>
 </slots>
</ui>
